src/AppWrapper.js

```diff
@@ -45,46 +45,52 @@ const providerDataTransformation = (rawData) => {
     }
 }
 
-const AppWrapper = () => {
-    const engine = useDataEngine()
-    const store = configureStore([
-        thunk.withExtraArgument(engine), // FIXME: Not needed for ER? Pointed out by @edoardo
-        metadataMiddleware,
-    ])
-
-    if (window.Cypress) {
-        window.store = store
-    }
-
+const AppWrapperInner = ({ engine }) => {
     const [ouLevels, setOuLevels] = useState(null)
-
     const doFetchOuLevelsData = useCallback(async () => {
         const ouLevels = await apiFetchOrganisationUnitLevels(engine)
-
         return ouLevels
     }, [engine])
 
     useEffect(() => {
         const doFetch = async () => {
             const ouLevelsData = await doFetchOuLevelsData()
-
             setOuLevels(ouLevelsData)
         }
-
         doFetch()
     }, [])
 
+    return (
+        <CachedDataQueryProvider
+            query={query}
+            dataTransformation={providerDataTransformation}
+        >
+            <App
+                initialLocation={history.location}
+                ouLevels={ouLevels} // TODO: Unused by App.js?
+            />
+        </CachedDataQueryProvider>
+    )
+}
+
+AppWrapperInner.propTypes = {
+    engine: PropTypes.object,
+}
+
+const AppWrapper = () => {
+    const engine = useDataEngine()
+    const store = configureStore([
+        thunk.withExtraArgument(engine), // FIXME: Not needed for ER? Pointed out by @edoardo
```

Comment:
I don't think I pointed out this one.
Although it should not be needed once all fetch functions in `api` are not used anymore.